lua/nvim-tree/git/init.lua

```diff
@@ -29,8 +29,8 @@ function M.reload_project(project_root, path)
     return
   end
 
-  if path and not path:match("^" .. project_root) then
-    path = nil
+  if path and path:find(project_root, 1, true) ~= 1 then
```

Comment:
This was pattern matching :facepalm: 

Resolves https://github.com/kyazdani42/nvim-tree.lua/pull/1373#discussion_r907919332